File: hw/maisincho/u3/HW20Compiste/CompositePy/controller/Employee.py
class Employee:
    def __init__(self, name, title):
        self.name = name
        self.title = title
        self.subordinates = [] 

    def add(self, employee):
        self.subordinates.append(employee)

    def remove(self, employee):
        if employee in self.subordinates:
            self.subordinates.remove(employee)

    def state_name(self):
        print(f"{self.title} {self.name}")
        for e in self.subordinates:
            e.state_name()

File: hw/maisincho/u3/HW20Compiste/Composite/src/main/java/ec/edu/espe/composite/controller/Employee.java
package ec.edu.espe.composite.controller;

import java.util.ArrayList;

/**
 *
 * @author Isaac Maisincho Crafters_Market DCCO ESPE
 */
public abstract class Employee {

    public String name;
    public String title;
    private ArrayList<Employee> subordinates = new ArrayList<>();

    public void add(Employee e) {
        subordinates.add(e);
    }

    public void remove(Employee e) {
        subordinates.remove(e);

    }

    public void stateName() {
        System.out.println(title + " " + name);
        for (Employee e : subordinates) {
            e.stateName();
        }
    }
}
